src/routing/stage.rs

```diff
@@ -145,18 +154,13 @@ impl Stage {
         delivery_group_size: usize,
         message: Bytes,
     ) -> Vec<Command> {
-        match self
-            .comm
+        self.comm
             .send_message_to_targets(recipients, delivery_group_size, message)
             .await
-        {
-            Ok(()) => vec![],
-            Err(error) => error
-                .failed_recipients
-                .into_iter()
-                .map(Command::HandlePeerLost)
-                .collect(),
-        }
+            .1
+            .into_iter()
+            .map(Command::HandlePeerLost)
```

Comment:
at this point, I think we shall only calll `HandleConnectionLost` ?
or, call `HandlePeerLost` only when the message is a `Ping` ?

or, you mean if the ConnectionLost is caused during a message lost, we can immediately claim that peer is lost ?

Reply:
>  if an error happend during sending a message, we can immediately claim that peer is lost

Yes, that's the idea.